code/modules/hydroponics/seed_datums.dm

```diff
@@ -532,6 +533,12 @@ var/global/list/gene_tag_masks = list()   // Gene obfuscation for delicious tria
 			//product.set_light(1+round(potency/50))
 			product.set_light(2)
 
+		//Clovers:
+		if(istype(product, /obj/item/weapon/reagent_containers/food/snacks/grown/clover))
```

Comment:
you should do this in a more OOP way; for example you could make a new after_harvest() proc

Reply:
Would `after_harvest()` be defined on `/obj/item/weapon/reagent_containers/food/snacks/grown`, `/datum/seed`, or something else? Doing it on `/datum/seed` wouldn't work I think because that would break for mutant plants with new seed datums. Since non-`/food/snacks/grown` products like dionaea are also possible, am I barking up the right tree in thinking it'd be something like:
```
if(istype(product, /obj/item/weapon/reagent_containers/food/snacks/grown))
     var/obj/item/weapon/reagent_containers/food/snacks/grown/G = product
     G.after_harvest()
```
?

Reply:
Oh maybe there's a better way.

Reply:
Or actually it can be done with `initialize()` i think.

Reply:
Or maybe it can work with the existing `initialize()` proc.

Reply:
Yes, but you make a good point in that this is only marginally better. I was envisioning putting it on the seed datum for the clover and having it operate on the product, but I see your point with the creation of a new datum for mutant plants. You could make it so that the creation of a new mutant passes the same seed datum type as its source type though.

Have a crack at that and if it becomes overwhelmingly annoying you can just implement it at the grown level as you suggest.

Reply:
Putting it in `grown/clover/initialize()` allowed me to simplify it a good bit.